binance_smart_chain/dex/trades/insert_dodo.sql

```diff
@@ -0,0 +1,312 @@
+            "toToken" AS token_b_address,
+            contract_address exchange_contract_address,
+            evt_tx_hash AS tx_hash,
+            NULL::integer[] AS trace_address,
+            evt_index
+        FROM
+            dodoex."DODOV2Proxy02_evt_OrderHistory"
+
+        UNION ALL
+
+        -- dodov2 dvm
+        SELECT
+            evt_block_time AS block_time,
+            'DODOEX' AS project,
+            'v2_bsc' AS version,
+            'DEX' AS category,
+            trader AS trader_a,
+            receiver AS trader_b,
+            "fromAmount" token_a_amount_raw,
+            "toAmount" token_b_amount_raw,
+            NULL::numeric AS usd_amount,
+            "fromToken" AS token_a_address,
+            "toToken" AS token_b_address,
+            contract_address exchange_contract_address,
+            evt_tx_hash AS tx_hash,
+            NULL::integer[] AS trace_address,
+            evt_index
+        FROM
+            dodoex."DVM_evt_DODOSwap"
+        WHERE trader <> '\xa356867fdcea8e71aeaf87805808803806231fdc'
+
+        UNION ALL
+
+        -- dodov2 dpp
+        SELECT
+            evt_block_time AS block_time,
+            'DODOEX' AS project,
+            'v2_bsc' AS version,
+            'DEX' AS category,
+            trader AS trader_a,
+            receiver AS trader_b,
+            "fromAmount" AS token_a_amount_raw,
+            "toAmount" AS token_b_amount_raw,
+            NULL::numeric AS usd_amount,
+            "fromToken" AS token_a_address,
+            "toToken" AS token_b_address,
+            contract_address AS exchange_contract_address,
+            evt_tx_hash AS tx_hash,
+            NULL::integer[] AS trace_address,
+            evt_index
+        FROM
+            dodoex."DPP_evt_DODOSwap"
+        WHERE trader <> '\xbe60d4c4250438344bec816ec2dec99925deb4c7'
+
+        UNION ALL
+
+        -- dodov2 dsp
+        SELECT
+            evt_block_time AS block_time,
+            'DODOEX' AS project,
+            'v2_bsc' AS version,
+            'DEX' AS category,
+            trader AS trader_a,
+            receiver AS trader_b,
+            "fromAmount" AS token_a_amount_raw,
+            "toAmount" AS token_b_amount_raw,
+            NULL::numeric AS usd_amount,
+            "fromToken" AS token_a_address,
+            "toToken" AS token_b_address,
+            contract_address AS exchange_contract_address,
+            evt_tx_hash AS tx_hash,
+            NULL::integer[] AS trace_address,
+            evt_index
+        FROM
+            dodoex."DSP_evt_DODOSwap"
+        WHERE trader <> '\xbe60d4c4250438344bec816ec2dec99925deb4c7'
+    ) dexs
+    INNER JOIN bsc.transactions tx
+        ON dexs.tx_hash = tx.hash
+        AND tx.block_time >= start_ts
+        AND tx.block_time < end_ts
+        AND tx.block_number >= start_block
+        AND tx.block_number < end_block
+    LEFT JOIN bep20.tokens erc20a ON erc20a.contract_address = dexs.token_a_address
```

Comment:
when you join this table, it's used to pull token symbol and use the decimal values for calculating raw token amount. i've noticed that the dodoex decoded tables have a distinct count of token addresses which is greater than the number found after the join. this causes null values for token symbols and raw amounts in end result. you may need to add all distinct token addresses (both a and b) from the decoded tables to this file:
https://github.com/duneanalytics/abstractions/blob/master/binance_smart_chain/bep20/tokens.sql

Reply:
reference query:
https://dune.com/queries/956021

switch between left and inner join helps show difference

Reply:
I deleted some of the same symbols. the contract_address should not be duplicate entries.

Reply:
Sorry, I didn't notice the case. Now I've updated it again.

Reply:
looks good!